### PR TITLE
Nav buttons and TOC style

### PR DESCRIPTION
**Pull request by davidfarmer**
Some tweaks so the navbar and toc look as expected.  Also added a link to the index.

Just a start.

### CODE REVIEW

File: src/components/index-button.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+export function IndexButton() {
+    const toolbar = useToolbarState();
+    const navTargets = useAppSelector(nextPrevParentSelector);
+    const dispatch = useAppDispatch();
+
+    return (
+            <ToolbarItem
+                {...toolbar}
+                as="a"
+                href="index-1.html"
```
Comment (siefkenj) [inline src/components/index-button.tsx:16]: I don't like hardcoded URLs...Is the index in the manifest? This should conditionally render depending on whether an index exists, right?

File: src/components/index-button.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                className="index-button button"
+                title="Index"
+                onClick={() => {
+                        dispatch(
+                            navActions.setCurrentPage("index-1")
```
Comment (siefkenj) [inline src/components/index-button.tsx:21]: Is it always called `index-1` and never `index`?

File: src/components/toc/index.tsx [hunk 3]
```diff
@@ -131,8 +131,8 @@ function TocEntry({ entry }: { entry: TocEntryType }) {
                 <Accordion open={openState}>
                     <ul
                         className={classNames({
-                            [`division-level-${entry.level}`]: entry.level,
-                        })}
+                            [`division-level-${entry.level}`]: entry.level
+                        }) + " structural"}
```
Comment (siefkenj) [inline src/components/toc/index.tsx:135]: `classNames` is magic. You don't need string concatenation. You can either do

```
classNames({ [`division-level-${entry.level}`]: entry.level }, "structural")
```
or

```
classNames({ [`division-level-${entry.level}`]: entry.level, structural: true })
```

**Comment by siefkenj** [comment]
Lookin' good! Are you hoping to merge this soon or are you wanting to work on it for a while?

**Comment by davidfarmer** [comment]
My preference is to merge it.  I already submitted two issues
about things which should be changed/done properly.  I think it
is easier to fix those after a merge, since those are not holding
up other work.

**Pull request merged**